src/publishPackages/publish.proj

```diff
@@ -53,33 +55,84 @@
+      <!-- Create a temporary directory to store the generated asset manifest by the task -->
+      <TempWorkingDirectory>$(ArtifactsDir)AssetsTmpDir\$([System.Guid]::NewGuid())/</TempWorkingDirectory>
     </PropertyGroup>
     <ItemGroup>
       <PackagesToPublish Include="$(ArtifactsPkgDir)**/*.nupkg" />
     </ItemGroup>
+    <MakeDir Directories="$(TempWorkingDirectory)"/>
+    <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] DB1 -- :: Files to Publish @(PackagesToPublish) " />
+    <PushToAzureDevOpsArtifacts
+      ItemsToPush="@(PackagesToPublish)"
+      ManifestBuildData="Location=$(ExpectedFeedUrl)"
+      ManifestRepoUri="$(ManifestName)"
+      ManifestBranch="$(ManifestBranch)"
+      ManifestBuildId="$(ManifestBuildId)"
+      ManifestCommit="$(ManifestCommit)"
+      IsStableBuild="false"
+      AssetManifestPath="$(AssetManifestFilePath)"
+      AssetsTemporaryDirectory="$(TempWorkingDirectory)" />
+
+    <!-- Copy the generated manifest to the build's artifacts -->
+    <Copy
+      SourceFiles="$(AssetManifestFilePath)"
+      DestinationFolder="$(TempWorkingDirectory)$(AssetManifestFileName)" />
+
+    <Message
+      Text="##vso[artifact.upload containerfolder=AssetManifests;artifactname=AssetManifests]$(TempWorkingDirectory)$(AssetManifestFileName)"
+      Importance="high" />
+
   </Target>
 
-  <Target Name="PreparePublishFiles" DependsOnTargets="PreparePublish;GetTarballFileName" BeforeTargets="PublishFilesToBlobFeed">
+  <Target Name="PreparePublishFiles" DependsOnTargets="PreparePublish;GetTarballFileName">
     <PropertyGroup>
-      <AssetManifestFilePath>$(ManifestDir)reference-packages-tar-manifest.xml</AssetManifestFilePath>
+      <AssetManifestFilename>reference-packages-tar-manifest.xml</AssetManifestFilename>
+      <AssetManifestFilePath>$(ManifestDir)$(AssetManifestFileName)</AssetManifestFilePath>
+      <!-- Create a temporary directory to store the generated asset manifest by the task -->
+      <TempWorkingDirectory>$(ArtifactsDir)AssetsTmpDir/$([System.Guid]::NewGuid())/</TempWorkingDirectory>
     </PropertyGroup>
     <ItemGroup>
       <FilesToPublish Include="$(TarballFileName)" />
     </ItemGroup>
-  </Target>
+    <MakeDir Directories="$(TempWorkingDirectory)"/>
+    <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] DB2 -- :: Files to Publish @(FilesToPublish) " />
+    <PushToAzureDevOpsArtifacts
+      ItemsToPush="@(FilesToPublish)"
+      ManifestBuildData="Location=$(ExpectedFeedUrl)"
+      ManifestRepoUri="$(ManifestName)"
+      ManifestBranch="$(ManifestBranch)"
+      ManifestBuildId="$(ManifestBuildId)"
+      ManifestCommit="$(ManifestCommit)"
+      PublishFlatContainer="true"
+      IsStableBuild="false"
+      AssetManifestPath="$(AssetManifestFilePath)"
+      AssetsTemporaryDirectory="$(TempWorkingDirectory)" />
 
+    <!-- Copy the generated manifest to the build's artifacts -->
```

Comment:
Depending on the version of Tasks.Feed that you're using you don't need to do this anymore.

Reply:
You mean we dont need to copy the generated manifest?

Reply:
Yeap. But you need to use the newest Tasks.Feed

Reply:
Depending on `MicrosoftDotNetBuildTasksFeedVersion` I expect it to be latest as far as we are on latest arcade? But as the subscription for ref-package repo is .NET3 we may not be? I am not sure about this though? Any thoughts on this?